src/sentry/utils/performance_issues/detectors/large_payload_detector.py

```diff
@@ -0,0 +1,95 @@
+from __future__ import annotations
+
+from sentry import features
+from sentry.issues.grouptype import PerformanceLargeHTTPPayloadGroupType
+from sentry.models import Organization, Project
+
+from ..base import DetectorType, PerformanceDetector, fingerprint_http_spans
+from ..performance_problem import PerformanceProblem
+from ..types import Span
+
+
+class LargeHTTPPayloadDetector(PerformanceDetector):
+    __slots__ = "stored_problems"
+
+    type: DetectorType = DetectorType.LARGE_HTTP_PAYLOAD
+    settings_key = DetectorType.LARGE_HTTP_PAYLOAD
+
+    def init(self):
+        self.stored_problems: dict[str, PerformanceProblem] = {}
+        self.consecutive_http_spans: list[Span] = []
+
+    def visit_span(self, span: Span) -> None:
+        span_id = span.get("span_id", None)
+
+        if not span_id or not self._is_eligible_http_span(span):
+            return
+
+        data = span.get("data", None)
+        encoded_body_size = data and data.get("Encoded Body Size", None)
+        if not (encoded_body_size):
+            return
+
+        payload_size_threshold = self.settings.get("payload_size_threshold")
+        if encoded_body_size > payload_size_threshold:
+            self._store_performance_problem(span)
+
+    def _store_performance_problem(self, span) -> None:
+        fingerprint = self._fingerprint(span)
+        offender_span_ids = span.get("span_id", None)
+        desc: str = span.get("description", None)
+
+        self.stored_problems[fingerprint] = PerformanceProblem(
+            fingerprint,
+            "http",
+            desc=desc,
+            type=PerformanceLargeHTTPPayloadGroupType,
+            cause_span_ids=[],
+            parent_span_ids=None,
+            offender_span_ids=offender_span_ids,
+            evidence_display=[],
+            evidence_data={
+                "parent_span_ids": [],
+                "cause_span_ids": [],
+                "offender_span_ids": offender_span_ids,
+                "op": "http",
+            },
+        )
+
+    def _is_eligible_http_span(self, span: Span) -> bool:
```

Comment:
Nit, but in other detectors this is a class method called `is_span_eligible`, would be good to keep it consistent!

Reply:
done!